save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ExecutionController.kt

```diff
@@ -139,15 +141,17 @@ class ExecutionController(private val executionService: ExecutionService,
      * @return ResponseEntity
      * @throws ResponseStatusException
      */
-    @PostMapping("/deleteAllExecution")
-    fun deleteExecutionForProject(@RequestParam name: String, @RequestParam owner: String): ResponseEntity<String>? {
-        val projectId = projectService.findByNameAndOwner(name, owner)?.id
-
+    @PostMapping("/execution/deleteAll")
+    fun deleteExecutionForProject(@RequestParam name: String, @RequestParam owner: String): ResponseEntity<String> {
         try {
-            testExecutionService.deleteTestExecutionWithProjectId(projectId)
-            agentService.deleteAgentWithProjectId(projectId)
-            executionService.deleteExecutionByProjectNameAndProjectOwner(name, owner)
+            projectService.findByNameAndOwner(name, owner)?.id?.let {
```

Comment:
Currently you are silently ignoring null values here. If project is not found (i.e. `find...` returns null), you should also return 404. You can achieve this by, for example, wrapping `find` in `requireNotNull`, which too throws `IllegalArgumentException`. And if project is found, then its `id` is guaranteed to be not null, so you can `!!` it.

Reply:
Fixed